api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -87,111 +88,154 @@
-          ResearchPurpose researchPurpose = new ResearchPurpose()
-              .diseaseFocusedResearch(workspace.getDiseaseFocusedResearch())
-              .diseaseOfFocus(workspace.getDiseaseOfFocus())
-              .methodsDevelopment(workspace.getMethodsDevelopment())
-              .controlSet(workspace.getControlSet())
-              .aggregateAnalysis(workspace.getAggregateAnalysis())
-              .ancestry(workspace.getAncestry())
-              .commercialPurpose(workspace.getCommercialPurpose())
-              .population(workspace.getPopulation())
-              .populationOfFocus(workspace.getPopulationOfFocus())
-              .additionalNotes(workspace.getAdditionalNotes())
-              .reviewRequested(workspace.getReviewRequested())
-              .approved(workspace.getApproved());
-
-          if(workspace.getTimeRequested() != null){
-            researchPurpose.timeRequested(workspace.getTimeRequested().getTime());
-          }
+          ResearchPurpose researchPurpose = createResearchPurpose(workspace);
 
-          if(workspace.getTimeReviewed() != null){
-            researchPurpose.timeReviewed(workspace.getTimeReviewed().getTime());
-          }
-          Workspace result = new Workspace()
-              .etag(Etags.fromVersion(workspace.getVersion()))
-              .lastModifiedTime(workspace.getLastModifiedTime().getTime())
-              .creationTime(workspace.getCreationTime().getTime())
-              .dataAccessLevel(workspace.getDataAccessLevel())
-              .name(workspace.getName())
-              .id(workspaceId.getWorkspaceName())
-              .namespace(workspaceId.getWorkspaceNamespace())
-              .description(workspace.getDescription())
-              .researchPurpose(researchPurpose);
-          if (workspace.getCreator() != null) {
-            result.setCreator(workspace.getCreator().getEmail());
-          }
-          if (workspace.getCdrVersion() != null) {
-            result.setCdrVersionId(String.valueOf(workspace.getCdrVersion().getCdrVersionId()));
-          }
 
+          if(workspace.getContainsUnderservedPopulation()) {
+            Set<UnderservedPopulationEnum> dbSet = workspace.getUnderservedPopulationSet();
+            List<UnderservedPopulationEnum> clientList = new ArrayList<UnderservedPopulationEnum>();
+            for (UnderservedPopulationEnum population : dbSet) {
+              clientList.add(population);
+            }
+            researchPurpose.setUnderservedPopulationDetails(clientList);
+          }
+          Workspace result = constructListWorkspaceFromDb(workspace, researchPurpose);
 
-          result.setUserRoles(workspace.getWorkspaceUserRoles().stream().map(TO_CLIENT_USER_ROLE).collect(Collectors.toList()));
+          return result;
+        }
+      };
 
+  @Deprecated
+  private static final Function<org.pmiops.workbench.db.model.Workspace, Workspace>
+      TO_CLIENT_WORKSPACE =
```

Comment:
Is this still in use?

Reply:
Yes, it is currently still in use. Adding comments and deleting the above one.